check-commit-email.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+
+# This script checks that the committer and author email addresses of commits
+# are valid - this is important because we need to be able to import the git
+# repo to launchpad and launchpad does not like git commits that are formatted
+# badly.
+# This can be run either on a merge commit in which case the commits it
+# evaluates is just limited to the "other" branch (the one being merged into the
+# destination) like we do in CI workflows. Or it can be run on a normal commit
+# like a developer would do locally in which case this ends up checking all
+# commits locally - some day that may grow to be unwieldy but for now seems ok.
+
+import subprocess
+from email.utils import parseaddr
+
+
+def get_commit_range():
+    # For CI, the head revision is a synthesised merge commit,
+    # merging the proposed branch into the destination branch.
+    # So the first parent is our destination, and the second is
+    # our proposal.
+    lines = subprocess.check_output(
+        ["git", "cat-file", "-p", "@"], text=True
+    ).splitlines()
+    parents = [
+        line[len("parent ") :].strip() for line in lines if line.startswith("parent ")
+    ]
+    if len(parents) == 1:
+        # not a merge commit, so return nothing to use default git log behavior
```

Comment:
I think this should cover the case when the branch cannot be merged to master due to conflicts

Reply:
It's also the case when you're running locally run-checks on a branch before proposing it